gradle-plugins/brownfield/src/main/kotlin/expo/modules/plugin/ExpoBrownfieldSetupPlugin.kt

```diff
@@ -0,0 +1,122 @@
+      appProject = findAppProject()
+
+      appBuildDir = appProject.layout.buildDirectory.get().asFile
+      moduleBuildDir = brownfieldProject.layout.buildDirectory.get().asFile
+
+      setupSourceSets()
+      setupCopyingAutolinking()
+      setupTasks()
+      setupCopyingNativeLibs()
+    }
+  }
+
+  private fun setupSourceSets() {
+    val main = libraryExtension.sourceSets.getByName("main")
+    main.java.srcDirs("$moduleBuildDir/generated/autolinking/src/main/java")
+
+    libraryExtension.sourceSets.getByName("release").apply {
+      jniLibs.srcDirs("libsRelease")
+      assets.srcDirs("$appBuildDir/generated/assets/createBundleReleaseJsAndAssets")
+      res.srcDirs("$appBuildDir/generated/res/createBundleReleaseJsAndAssets")
+    }
+  
+    libraryExtension.sourceSets.getByName("debug").apply {
+      jniLibs.srcDirs("libsDebug")
+    }
+  }
+
+  private fun setupCopyingAutolinking() {
+    val path = "generated/autolinking/src/main/java"
+    brownfieldProject.tasks.register("copyAutolinkingSources", Copy::class.java) { task ->
+      task.dependsOn(":${appProject.name}:generateAutolinkingPackageList")
+      task.from("$appBuildDir/$path")
+      task.into("$moduleBuildDir/$path")
+
+      val rnEntryPointTask = appProject.tasks.findByName("generateReactNativeEntryPoint")
+      if (rnEntryPointTask != null) {
+        task.dependsOn(rnEntryPointTask)
+      }
+
+      task.doLast {
+        val sourceFile = File(moduleBuildDir, "$path/com/facebook/react/ReactNativeApplicationEntryPoint.java")
+        if (sourceFile.exists()) {
+            var content = sourceFile.readText()
+            val namespace = libraryExtension.namespace 
+              ?: throw IllegalStateException("Namespace hasn't been configured for the library extension")
+            val regex = Regex("""\b[\w.]+(?=\.BuildConfig)""")
+            content = content.replace(regex, namespace)
+            sourceFile.writeText(content)
+        }
+      }
+    }
+  }
+
+  private fun setupTasks() {
+    brownfieldProject.tasks.named("preBuild").configure { task ->
+      task.dependsOn("copyAutolinkingSources")
+    }
+
+    brownfieldProject.tasks.named("preReleaseBuild").configure { task ->
+      task.dependsOn(appProject.tasks.named("createBundleReleaseJsAndAssets"))
+    }
+  }
+
+  private fun setupCopyingNativeLibs() {
+    setupCopyingNativeLibsForType("Release")
+    setupCopyingNativeLibsForType("Debug")
+  }
+
+  private fun setupCopyingNativeLibsForType(buildType: String) {
+    val mergeJniLibsTask = brownfieldProject.tasks.named("merge${buildType}JniLibFolders")
+
+    val stripTaskPath = ":${appProject.name}:strip${buildType}DebugSymbols"
+    val codegenTaskPath = ":${brownfieldProject.name}:generateCodegenSchemaFromJavaScript"
+
+    val fromDir = appProject.layout.buildDirectory
+      .dir("intermediates/stripped_native_libs/${buildType.toLowerCase()}/strip${buildType}DebugSymbols/out/lib")
+      .get().asFile
```

Comment:
```suggestion
    val fromDir = appProject.layout.buildDirectory
      .dir("intermediates/stripped_native_libs/${buildType.toLowerCase()}/strip${buildType}DebugSymbols/out/lib")
```
You should be able to pass `FileProvider` as a task input